activestorage/db/update_migrate/20180723000244_add_foreign_key_constraint_to_active_storage_attachments_for_blob_id.rb

```diff
@@ -1,6 +1,8 @@
 class AddForeignKeyConstraintToActiveStorageAttachmentsForBlobId < ActiveRecord::Migration[6.0]
   def up
-    unless foreign_key_exists?(:active_storage_attachments, column: :blob_id)
+    return if foreign_key_exists?(:active_storage_attachments, column: :blob_id)
+
+    if table_exists?(:active_storage_attachments) && table_exists?(:active_storage_blobs)
```

Comment:
These tables create with same migration file. So I think it's enough to check only one table.

```ruby 
    if table_exists?(:active_storage_attachments) && !foreign_key_exists?(:active_storage_attachments, column: :blob_id)
      add_foreign_key :active_storage_attachments, :active_storage_blobs, column: :blob_id
    end
```

Reply:
Hi @y-yagi, thanks for reviewing it, I do not know if I have explained myself well.

Updating to rails 6.0.0.beta3 from 6.0.0.beta2 if the command **`rake app:update`** is used a new migration file is generated ( **Even if active storage has not been installed yet in rails 6.0.0.beta2 or before and the tables do NOT exist** )

The command `app:update` will generate:

```ruby
# This migration comes from active_storage (originally 20180723000244)
class AddForeignKeyConstraintToActiveStorageAttachmentsForBlobId < ActiveRecord::Migration[6.0]
  def up
    unless foreign_key_exists?(:active_storage_attachments, column: :blob_id)
      add_foreign_key :active_storage_attachments, :active_storage_blobs, column: :blob_id
    end
  end
end
```

And this migration will causes the error:

> **ActiveRecord::StatementInvalid: PG::UndefinedTable: ERROR:  relation "active_storage_attachments" does not exist**

Because active storage is not installed yet and the table does not exist but the migration to add_foreign_key is generated for inexistent table

I tried to explain it in the issue #35621